### ISSUE TITLE
How to train Yolov5 custom model on custom Dataset using PYTORCH 

### ISSUE THREAD

**Issue by tanulsingh** [ultralytics/yolov5#390]
When I download yolov5 as torch hub model like
Model = torch.hub......
Now I want to use the model to find the predictions and the loss and backpropagate using my custom training function. I want to write custom training, evaluation and everything
I AM not able to understand what's the output of the model, will the model output the loss as well if I supply the targets along the image or I need to just supply the image and get the predictions then find the loss myself?
If so how??
A small code snippet for custom training will help a lot 
Thanks

**Comment by tonmoy-TS** [comment]
Was it solved?

**Comment by glenn-jocher** [comment]
@tanulsingh 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by tonmoy-TS** [comment]
Hi @glenn-jocher . Thanks for your reply. 
However, I guess both @tanulsingh and me were looking for an alternative for using train.py. i.e. is it possible to train a custom model using PyTorch Hub, just like you showed a way to inference in the above example?

**Comment by glenn-jocher** [comment]
@tonmoy-TS PyTorch Hub provides you a model, what you do with the model is up to you. 

You can build your own training code and use the model, sure.
